frontend/app/components/Reviews.tsx

```diff
@@ -0,0 +1,107 @@
+import { cn } from '@/lib/utils';
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+import { Review } from '@/app/review.types';
+
+export default function Reviews({ params }: { params: { movieId: string } }) {
+  const [reviews, setReviews] = useState<Review[]>([]);
+
+  useEffect(() => {
+    const fetchReviews = async () => {
+      const response = await fetch('/api/reviews');
+      const data = await response.json();
+      setReviews(data);
+    };
+    fetchReviews();
+  }, []);
+
+  // Get reviews for movie id
+  useEffect(() => {
+    if (params.movieId !== undefined) {
+      fetch(`http://localhost:8000/reviews/${params.movieId}`)
+        .then(res => res.json())
+        .then(data => setReviews(data));
+    }
+  }, [params.movieId]);
+
+  return (
+    <div className="bg-white">
+      <div>
+        <h2 className="sr-only">Customer Reviews</h2>
+        <div className="my-10">
+          {reviews.map((review, reviewIdx) => (
+            <div
+              key={review.id}
+              className="flex space-x-4 text-sm text-gray-500"
+            >
+              <div className="flex-none py-10">
+                <Image
+                  src="https://images.unsplash.com/photo-1551963831-b3b1ca40c98e"
+                  alt="reviewer avatar"
+                  className="h-10 w-10 rounded-full bg-gray-100"
+                  width={40}
+                  height={40}
+                />
+              </div>
+              <div
+                className={cn(
+                  reviewIdx === 0 ? '' : 'border-t border-gray-200',
+                  'flex-1 py-10'
+                )}
+              >
+                <h3 className="font-medium text-gray-900">
+                  {review.DisplayName}
+                </h3>
+                <p>
+                  <time dateTime={review.Date}>
+                    {new Date(review.Date).toLocaleDateString('en-US', {
+                      year: 'numeric',
+                      month: 'long',
+                      day: 'numeric',
+                      hour: 'numeric',
+                      minute: 'numeric',
+                      timeZoneName: 'short'
+                    })}
+                  </time>
+                </p>
+                <div className="mt-4 flex items-center">
+                  {[0, 1, 2, 3, 4].map(rating => (
+                    <div key={rating} className="relative">
+                      <StarIcon
+                        className={cn(
+                          review.stars > rating
+                            ? 'text-yellow-400'
+                            : 'text-gray-300',
+                          'h-5 w-5 flex-shrink-0'
+                        )}
+                        aria-hidden="true"
+                      />
+                      {review.stars > rating && review.stars < rating + 1 && (
+                        <StarIcon
+                          className="absolute left-0 top-0 h-5 w-5 flex-shrink-0"
```

Comment:
The logic here for displaying half stars seems really complicated and could have performance issues with clipping. Instead of dynamically clipping the star, we could have a half star and full star icon. Then here, you could just call a render function to display the right stars. That way we dont have complicated logic inside of the JSX itself. I'm thinking something like this:


```jsx
const generateStars = (rating) => {
  const stars = [];
  for (let i = 0; i < 5; i++) {
    if (rating > i) {
      if (rating < i + 1) {
        // Half star logic
        stars.push(
          <HalfStarIcon key={i} className="text-yellow-400 h-5 w-5 flex-shrink-0" aria-hidden="true" />
        );
      } else {
        // Full star
        stars.push(
          <StarIcon key={i} className="text-yellow-400 h-5 w-5 flex-shrink-0" aria-hidden="true" />
        );
      }
    } else {
      // Empty star
      stars.push(
        <StarIcon key={i} className="text-gray-300 h-5 w-5 flex-shrink-0" aria-hidden="true" />
      );
    }
  }
  return stars;
};
```

And then you can just replace lines 80 through 91 with `generateStars(rating)` This would be much easier on the browser side because there are no pixel and subpixel calculations to determine what a half star is.

Reply:
Sounds good, will try that

Reply:
There wasn't an existing `HalfStarIcon` so I just created my own. Used the clipPath to create the effect though, I couldn't figure out how to do it with other CSS properties, not CSS guru enough.

Reply:
Check this out:

https://www.npmjs.com/package/@heroicons/react

Reply:
Wait nvm

Reply:
I just made this one. It's around the same performance as the one you built but doesn't use the style prop.

```jsx
const HalfStarIcon = () => (
  <div className="relative h-5 w-5 flex-shrink-0">
    <StarIcon
      className="absolute inset-0 h-full w-full text-gray-300"
      aria-hidden="true"
    />
    <div className="absolute inset-0 h-full w-1/2 overflow-hidden">
      <StarIcon className="h-full w-full text-yellow-400" aria-hidden="true" />
    </div>
  </div>
);
```

Reply:
Found a person who made an SVG of it here
https://tailwindflex.com/@prajwal/star-rating-svg-buttons, we could also try that and just have 

```
function HalfStarIcon() {
  return (
    <svg>
        rest of code for svg here
    </svg>
  );
}
```

Reply:
Jk the shape of his stars are off